ddspipe_yaml/src/cpp/YamlReader_types.cpp

```diff
@@ -461,19 +467,26 @@ void YamlReader::fill(
         const Yaml& yml,
         const YamlReaderVersion version)
 {
-    // Optional name
-    if (is_tag_present(yml, TOPIC_NAME_TAG))
-    {
-        object.topic_name.set_value(get<std::string>(yml, TOPIC_NAME_TAG, version));
-    }
+    // Name required
+    object.topic_name = get<std::string>(yml, TOPIC_NAME_TAG, version);
 
     // Optional data type
     if (is_tag_present(yml, TOPIC_TYPE_NAME_TAG))
     {
         object.type_name.set_value(get<std::string>(yml, TOPIC_TYPE_NAME_TAG, version));
     }
 
-    // TODO: decide whether we want to use QoS as filtering
+    // Optional QoS
+    if (is_tag_present(yml, TOPIC_QOS_TAG))
+    {
+        fill<TopicQoS>(object.topic_qos, get_value_in_tag(yml, TOPIC_QOS_TAG), version);
+    }
+
+    // Optional participants tag
+    if (is_tag_present(yml, TOPIC_PARTICIPANTS_TAG))
+    {
+        object.participants.set_value(get_set<ParticipantId>(yml, TOPIC_PARTICIPANTS_TAG, version));
+    }
```

Comment:
`TOPICS_TAG` parsing missing

Reply:
The `TOPICS_TAG` is parsed in the parent. In the case of the router it is parsed in the YamlReader_configuration. Although, now that you mention it, it may be interesting to have a YamlReader_configuration in the DdsPipe that parses the DdsPipeConfiguration.

Reply:
The problem would be that, right now, the structure of the Yaml is not required to be the same in the different projects that use the DdsPipe.